src/core/common/message.hpp

```diff
@@ -229,10 +229,10 @@ class Message : public Buffer
 
     enum
     {
-        kPriorityHigh    = 0, ///< High priority level.
-        kPriorityMedium  = 1, ///< Medium priority level.
-        kPriorityLow     = 2, ///< Low priority level.
-        kPriorityVeryLow = 3, ///< Very low priority level.
+        kPriorityNetworkControl = 0,                          ///< Network Control priority level.
```

Comment:
Why not just use `otMessagePriority` instead of creating another internal enum?
Also, shouldn't `otMessagePriority` explicitly reserve the value `0` to avoid any conflicts with this internal definition?

Reply:
The priority kPriorityNetworkControl is only used by the internal network control messages. We don't want the user set their application message to priority kPriorityNetworkControl. So I did't expose the kPriorityNetworkControl in the enum `otMessagePriority` and use `otMessagePriority`  as the internal enum.

You are right. I will add priority OT_MESSAGE_PRIORITY_RESERVED to the enum `otMessagePriority` to avoid conflicts.

Reply:
I reversed the priority value. So the priority value 0 indicates priority OT_MESSAGE_PRIORITY_LOW. Please review the latest commit.